### ISSUE TITLE
Learning to Sample from Complex Distribution (High-Dimensional)

### ISSUE THREAD

**Issue by kayuksel** [tensorflow/probability#1253]
Hello everyone,

I would like a NN to learn how to sample from a complex distribution in high-dimensions (e.g. 10-100K dimensions).

I guess that would be possible via learning parameters of a multivariate normal distribution e.g. MultivariateNormalTriL

However, in high-dimensional problems, the parameter-size becomes too much and this becomes memory inefficient.

Can you please suggest to me an alternative if it exists in the framework or another method in order for achieving this?

Note: The scalability of the method is the most crucial aspect for me, the resulting expressiveness is not that important.

Thank you very much.

Sincerely,
Kamer

**Comment by srvasude** [comment]
Hi,
   I guess the set up you have is a NN that outputs a cholesky factor and mean of a Normal distribution, and you would like to sample from that? 
   
  We have a `MultivariateNormalLinearOperator` which can take a `LinearOperator` for the scale factor. You can choose a `LinearOperator` such as `LinearOperatorDiag`. This would result in both a smaller parameterization than the lower triangular distribution, as well as faster sampling (since inverting a diagonal matrix is cheaper than a triangular solve). More generally, you can choose an operator that has a fast solve / doesn't require that many parameters to sample fast. The downside of all this is you might not capture correlations between different outputs.

**Comment by kayuksel** [comment]
@srvasude Thank you very much. At the moment, it outputs mean and std of normal distribution but I can change that as it needs.
I will check about MultivariateNormalLinearOperator and LinearOperatorDiag. Would that be the same as MultivariateNormalDiag?

**Comment by srvasude** [comment]
Yes, that would be the same. I mentioned `LinearOperatorDiag` since you could swap it out with other operators as listed here: https://www.tensorflow.org/api_docs/python/tf/linalg/LinearOperator (for instance, depending on the problem, you could decompose as a kronecker product which will be more memory efficient / faster solves).
